### ISSUE TITLE
Radio: a11y: readOnly state should be revisited 

### ISSUE THREAD

**Issue by jbadan** [SAP/fundamental-styles#1769]
readOnly is not a supported for Radio inputs  - https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly

![Screen Shot 2020-10-09 at 6 41 05 AM](https://user-images.githubusercontent.com/29607818/95590345-ca926d80-09fa-11eb-99ff-ad0ff1914848.png)

"The attribute is not supported or relevant to <select> or input types that are already not mutable, such as checkbox and radio or cannot, by definition, start with a value, such as the file  input type. range and color, as both have default values."

"Note: Only text controls can be made read-only, since for other controls (such as checkboxes and buttons) there is no useful distinction between being read-only and being disabled, so the readonly attribute does not apply."

@jacobdevera @meganaconley @Mike-Diaz @prsdthkr Could you confirm if my understanding of this is correct?

**Comment by jacobdevera** [comment]
This is correct, although there is a read-only style defined in Fiori: https://experience.sap.com/fiori-design-web/radio-button/#states

This would need to be removed by UX unless there really is a case for allowing read-only buttons and not just a UI state that was blindly added for completeness.

**Comment by salarenko** [comment]
@jbadan @jacobdevera 
I'm not sure if I agree with you. There are two things we are loosing when we decide to abandon `read-only` and use `disabled` instead:

- User won't be able to focus form control as it should happen for read-only controls (won't this cause some accessibility issues?)
- Disabled controls are not submitted with the form

Are we going to somehow handle these cases?

**Comment by jacobdevera** [comment]
Neither `readonly` nor `aria-readonly` are supported by `<input type="radio">` nor any other related ARIA role. This is an inherent limitation of the HTML / ARIA spec and we wouldn't be able to get a "readonly" radio to be recognized properly by assistive technologies.

If the desire is simply to keep the "read-only" styling from Fiori, a workaround could be to set the other radio options to `disabled` in the markup and leave the selected option as enabled, while styling them all with the "read-only" styling. Again though, there would be no way to convey this as `readonly` to non-sighted users.
